src/commands.ts

```diff
@@ -295,11 +290,37 @@ export class Commands
         }
     }
 
+    async applySettings(f) : Promise<void> {
+        let target = await this.pickConfigTarget();
+
+        if (target === undefined) {
+            return;
+        }
+
+        try {
+            f(target);
+        }
+        catch(err) {
+            this._vscode.showErrorMessage(err.message);
+        }
+    }
+
     register()
     {
         this._vscode.registerAndSubscribeCommand('rock.updatePackageInfo', () => { this.updatePackageInfo() });
         this._vscode.registerAndSubscribeCommand('rock.addLaunchConfig', () => { this.addLaunchConfig() });
-        this._vscode.registerAndSubscribeCommand('rock.updateCodeConfig', () => { this.updateCodeConfig() });
+        this._vscode.registerAndSubscribeCommand('rock.applyDefaultSettings', () => {
+            this.applySettings((target) => this._configManager.applyDefaultSettings(target));
+        });
+        this._vscode.registerAndSubscribeCommand('rock.applyCPPSettings', () => {
```

Comment:
Since you are at this.. vscode-cpptools now allows having what was previously set in c_cpp_properties.json globally in a workspace. We are creating that file only to set the compile_commands.json path so what do you think about moving that to the workspace scope as well?

Reply:
Sounds good ... but tbh I'd rather have this merged and then work on improving the rest.

The only thing I could find about that is this: https://github.com/Microsoft/vscode-cpptools/blob/master/Documentation/LanguageServer/Customizing%20Default%20Settings.md. I'm really not sure how it is supposed to work, to provide a different json file for each folder ?

Reply:
> but tbh I'd rather have this merged and then work on improving the rest

Sound good...

> to provide a different json file for each folder ?

```typescript
C_Cpp.default.compileCommands: "${workspaceFolder}/{config:rock.buildDir}/compile_commands.json"
```

Works for relative build dirs... I just realized that we have to support absolute build dirs as well and best thing I can think of right now is to fallback to c_cpp_properties.json in this case.

Reply:
Never mind.. Won't work for the your multi-workspace use case anyway..

Reply:
What's `{config:rock.buildDir}` in here ? How is it provided ?

Reply:
https://github.com/Microsoft/vscode-cpptools/blob/3d7937feeb1699492969acb6c8e6b915cd4abfae/Extension/src/common.ts#L317

The extension would have to set that according to `.autoproj/config.yml` on the workspace scope configuration. Won't work for multi workspaces though so the whole point is moot